packages/data/src/items/selectors.js

```diff
@@ -1,19 +1,32 @@
+/**
+ * External dependencies
+ */
+import createSelector from 'rememo';
+
 /**
  * Internal dependencies
  */
 import { getResourceName } from '../utils';
 import { getTotalCountResourceName } from './utils';
 
-export const getItems = ( state, itemType, query ) => {
-	const resourceName = getResourceName( itemType, query );
-	const ids =
-		( state.items[ resourceName ] && state.items[ resourceName ].data ) ||
-		[];
-	return ids.reduce( ( map, id ) => {
-		map.set( id, state.data[ itemType ][ id ] );
-		return map;
-	}, new Map() );
-};
+export const getItems = createSelector(
+	( state, itemType, query, defaultValue = new Map() ) => {
+		const resourceName = getResourceName( itemType, query );
+		const ids =
+			state.items[ resourceName ] && state.items[ resourceName ].data;
+		if ( ! ids ) {
+			return defaultValue;
+		}
+		return ids.reduce( ( map, id ) => {
+			map.set( id, state.data[ itemType ][ id ] );
+			return map;
+		}, new Map() );
+	},
+	( state, itemType, query ) => {
+		const resourceName = getResourceName( itemType, query );
+		return [ state.items[ resourceName ] ];
+	}
+);
```

Comment:
Memoized this as we are creating a new array on every call. I also added a `defaultValue` (mostly for backwards compatibility).
Mostly to handle the loading state correctly and to know when it was set to an empty array or still `undefined`. Given the `isRequesting` selector had a slight delay that was noticeable in the UI.